### PR TITLE
Private channel payload

### PR DESCRIPTION
**Pull request by jweslley**
Pusher accepts js clients to use the same auth endpoint to both presence and private channels.

### CODE REVIEW

File: ipe/websockets.go [hunk 1]
```diff
@@ -123,13 +123,13 @@ func onMessage(conn *websocket.Conn, w http.ResponseWriter, r *http.Request, ses
...
-			isPresence := strings.HasPrefix(channelName, "presence-")
-			isPrivate := strings.HasPrefix(channelName, "private-")
+			isPresence := utils.IsPresenceChannel(channelName)
+			isPrivate := utils.IsPrivateChannel(channelName)
 
 			if isPresence || isPrivate {
 				toSign := []string{connection.SocketID, channelName}
 
-				if isPresence {
+				if isPresence || len(subscribeEvent.Data.ChannelData) > 0 {
```
Comment (dimiro1) [inline ipe/websockets.go]: Why are you checking the length of the ChannelData string?

**Comment by dimiro1** [comment]
This PR is almost equal the last one you open[1]. I already merged the another [1], please rebase your branch.

[1] https://github.com/dimiro1/ipe/pull/6

File: ipe/websockets.go (hunk 1 above)
Comment (jweslley) [inline ipe/websockets.go]: When authenticating a private channel, it is possible to send some payload containing user info, just like presence channels. Thus, if some payload is provided during private channel's authentication, this payload (ChannelData) must be used to generate the auth key.

**Comment by jweslley** [comment]
Sorry. my fault. I'll rebase. ^^

File: ipe/websockets.go (hunk 1 above)
Comment (dimiro1) [inline ipe/websockets.go]: Ok Looks fine.

**Pull request merged**